Zend/zend_vm_def.h

```diff
@@ -8091,6 +8095,74 @@ ZEND_VM_HANDLER(149, ZEND_HANDLE_EXCEPTION, ANY, ANY)
 	ZEND_VM_DISPATCH_TO_HELPER(zend_dispatch_try_catch_finally_helper, try_catch_offset, current_try_catch_offset, op_num, throw_op_num);
 }
 
+ZEND_VM_HANDLER(204, ZEND_HANDLE_DELAYED_ERROR, ANY, ANY)
+{
+	const zend_op *prev_op = EG(opline_before_exception);
+	bool delay = false;
+	switch (prev_op->opcode) {
+		case ZEND_FETCH_W:
+		case ZEND_FETCH_RW:
+		case ZEND_FETCH_FUNC_ARG:
+		case ZEND_FETCH_UNSET:
+		case ZEND_FETCH_DIM_W:
+		case ZEND_FETCH_DIM_RW:
+		case ZEND_FETCH_DIM_UNSET:
+		case ZEND_FETCH_LIST_W:
+		case ZEND_FETCH_OBJ_W:
+		case ZEND_FETCH_OBJ_RW:
+		case ZEND_FETCH_OBJ_UNSET:
+			delay = true;
+			break;
+	}
+
+	// FIXME: Is this guaranteed to be there?
+	const zend_op *next_op = prev_op + 1;
+	if (next_op->opcode == ZEND_OP_DATA) {
+		next_op++;
+	}
+
+	if (delay) {
+		zend_op *delayed_op = &EG(delayed_error_op)[0];
+		*delayed_op = *next_op;
```

Comment:
In case you have few ``FETCH`` instructions in row and few of them produce warnings, then EG(delayed_error_op)[0] is going to be overridden few times. Is it OK?
Will this work in conjunction with magic __get() and different warnings in main code and magic methods?

Reply:
> In case you have few FETCH instructions in row and few of them produce warnings, then EG(delayed_error_op)[0] is going to be overridden few times. Is it OK?

Yes. The idea is that each instruction producing indirect values only delays to the next opcode. If that opcode also produces an indirect value, then it wil once again set `EG(delayed_error_op)[0]` to the instruction after that, until eventually the indirect value is used and the error can be emitted.

Note that this happens only if the first FETCH emits a warning. The error is still delayed until the fetch+assign chain.

>  Will this work in conjunction with magic __get() and different warnings in main code and magic methods?

I didn't test this, but I think one unexpected things could be that we don't associate `EG(delayed_errors)` with a particular stack frame. If the magic method produces a warning, pending warnings from the outer VM call will also be handled. This could be solved by storing the `execute_data` on the delayed error, and only handling those belonging to the current one. Apart from that I believe magic methods should work correctly with this approach.